src/rebalancers/uniswap-v3/UniswapV3Rebalancer.sol

```diff
@@ -0,0 +1,487 @@
+
+    // A struct with the state of a specific position, only used in memory.
+    struct PositionState {
+        address pool;
+        address token0;
+        address token1;
+        uint24 fee;
+        int24 newUpperTick;
+        int24 newLowerTick;
+        uint128 liquidity;
+        uint256 sqrtPriceX96;
+        uint256 lowerBoundSqrtPriceX96;
+        uint256 upperBoundSqrtPriceX96;
+    }
+
+    // A struct used to store information for each specific initiator
+    struct InitiatorInfo {
+        uint256 upperSqrtPriceDeviation;
+        uint256 lowerSqrtPriceDeviation;
+        uint256 fee;
+    }
+
+    /* //////////////////////////////////////////////////////////////
+                                ERRORS
+    ////////////////////////////////////////////////////////////// */
+
+    error DecreaseFeeOnly();
+    error DecreaseToleranceOnly();
+    error FeeAlreadySet();
+    error LiquidityTresholdExceeded();
+    error MaxInitiatorFee();
+    error MaxTolerance();
+    error NotAnAccount();
+    error OnlyAccount();
+    error OnlyPool();
+    error Reentered();
+    error UnbalancedPool();
+    error InitiatorNotValid();
+
+    /* //////////////////////////////////////////////////////////////
+                                EVENTS
+    ////////////////////////////////////////////////////////////// */
+
+    event Rebalance(address indexed account, uint256 id);
+
+    /* //////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    ////////////////////////////////////////////////////////////// */
+
+    /**
+     * @param maxTolerance The maximum allowed deviation of the actual pool price for any initiator,
+     * relative to the price calculated with trusted external prices of both assets, with 18 decimals precision.
+     */
+    constructor(uint256 maxTolerance) {
+        MAX_TOLERANCE = maxTolerance;
+    }
+
+    /* ///////////////////////////////////////////////////////////////
+                             REBALANCING LOGIC
+    /////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice Rebalances a UniswapV3 Liquidity Position owned by an Arcadia Account.
+     * @param account_ The Arcadia Account owning the position.
+     * @param id The id of the Liquidity Position to rebalance.
+     * @param lowerTick The new lower tick to rebalance to.
+     * @param upperTick The new upper tick to rebalance to.
+     * @dev When both lowerTick and upperTick are zero, ticks will be updated with same tick-spacing as current position
+     * and with a balanced, 50/50 ratio around current tick.
+     */
+    function rebalancePosition(address account_, uint256 id, int24 lowerTick, int24 upperTick) external {
+        // Store Account address, used to validate the caller of the executeAction() callback.
+        if (account != address(0)) revert Reentered();
+        if (!ArcadiaLogic.FACTORY.isAccount(account_)) revert NotAnAccount();
```

Comment:
Check if it is a valid account only in setInitiatorForAccount. -> no need to check it on every rebalancePosition

Reply:
done